### PR TITLE
Add option to promote input variables to state variables

### PR DESCRIPTION
**Pull request by makaimann**

### CODE REVIEW

**Comment by makaimann** [comment]
Built this on https://github.com/upscale-project/pono/pull/213 so marking `do not merge` until that's merged.

**Comment by makaimann** [comment]
FYI @zhanghongce, once this is merged you could just require that this pass is run before running `SyGuS-PDR` -- or even just call `promote_inputvar` on all the input variables internally.

**Review by ahmed-irfan — APPROVED** [review]
LGTM. 

Please check the minor comments

File: core/ts.cpp
Comment (ahmed-irfan) [inline core/ts.cpp:397]: we have a similar assert above. Is this needed?

File: pono.cpp
Comment (ahmed-irfan) [inline pono.cpp:100]: can it be `auto const & iv`?

File: core/ts.cpp
Comment (makaimann) [reply core/ts.cpp]: Good point addressed with be917378730b7130df2ccc9959c8037134a79fe7

File: pono.cpp
Comment (makaimann) [reply pono.cpp]: Good point! Addressed with be917378730b7130df2ccc9959c8037134a79fe7.

**Pull request merged**
